src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/TextMapPropagator.cs

```
@@ -0,0 +1,139 @@
+
+        /// <summary>
+        /// Extracts the value from an incoming request represented by the carrier. For example, from the headers of an HTTP request.
+        /// </summary>
+        /// <param name="carrier">Carrier is the medium used by the propagators to read values from.</param>
+        /// <param name="getter">The callback will be invoked to get the propagation trace Id and trace state from carrier.</param>
+        /// <param name="traceId">The extracted trace Id from the carrier.</param>
+        /// <param name="traceState">The extracted trace state from the carrier.</param>
+        public abstract void ExtractTraceIdAndState(object carrier, PropagatorGetterCallback getter, out string? traceId, out string? traceState);
+
+        /// <summary>
+        /// Extracts the baggage key-value pairs list from an incoming request represented by the carrier. For example, from the headers of an HTTP request.
+        /// </summary>
+        /// <param name="carrier">Carrier is the medium used by the propagators to read values from.</param>
+        /// <param name="getter">The callback will be invoked to get the propagation baggage list from carrier.</param>
+        /// <returns>Returns the extracted key-value pair list from teh carrier.</returns>
+        public abstract IEnumerable<KeyValuePair<string, string?>>? ExtractBaggage(object carrier, PropagatorGetterCallback getter);
+
+        /// <summary>
+        /// Get or set the process wide propagator object which used as the current selected propagator.
+        /// </summary>
+        public static TextMapPropagator Current
+        {
+            get
+            {
+                Debug.Assert(s_current is not null);
+                return s_current;
+            }
+
+            set
+            {
+                if (value is not null)
+                {
+                    s_current = value;
+                }
```

```suggestion
                s_current = value ?? throw new ArgumentNullException(nameof(value));
```

---

I am trying to avoid throwing in the propagator APIs in general. so I prefer not throwing here. let me know if you think otherwise.

---

I would prefer we throw here to indicate to the user they did something we don't support.

If we have an api like the property on `SocketsHttpHandler` that is nullable and null has an actual meaning, it would be misleading to accept a null here.

---

ok, I'll let it throw then.